apis/python/src/tiledb/vector_search/object_api/object_index.py

```diff
@@ -29,12 +28,14 @@ def __init__(
         config: Optional[Mapping[str, Any]] = None,
         timestamp=None,
         load_embedding: bool = True,
+        load_metadata_in_memory: bool = True,
         **kwargs,
     ):
         with tiledb.scope_ctx(ctx_or_config=config):
             self.uri = uri
             self.config = config
             self.timestamp = timestamp
+            self.load_metadata_in_memory = load_metadata_in_memory
```

Comment:
(Non-blocking) What is the case where we will want to set `load_metadata_in_memory` to false? Is it about the memory pressure / speed? Curious if we should just always do this and not expose an option.

Reply:
For large datasets, for which the dataset can't fit in memory or we don't want to impose large memory requirements to the client, we want to have this processing happen from the TileDB array in an out-of-core manner. This is similar to how we perform vector search queries, we offer both memory based and out-of-core implementations.

Reply:
Makes sense, thanks.